src/cursor/cur_std.c

```diff
@@ -849,67 +865,199 @@ __wt_cursor_get_hash(
+    *readonlyp = readonly;
+    return (0);
+}
+
+/*
+ * __any_cursor_can_be_cached --
+ *     Determine whether cfg[] allows that a cursor be cached.
+ */
+static int
+__any_cursor_can_be_cached(WT_SESSION_IMPL *session, const char *cfg[], bool *cacheablep)
+{
+    WT_CONFIG_ITEM cval;
+    /*
+     * Any cursors that have special configuration cannot be cached. There are some exceptions for
+     * configurations that only differ by a cursor flag, which we can patch up if we find a matching
+     * cursor.
+     */
+    WT_RET(__wt_config_gets_def(session, cfg, "bulk", 0, &cval));
+    if (cval.val)
+        goto return_false;
+
+    WT_RET(__wt_config_gets_def(session, cfg, "debug", 0, &cval));
+    if (cval.len != 0)
+        goto return_false;
+
+    WT_RET(__wt_config_gets_def(session, cfg, "dump", 0, &cval));
+    if (cval.len != 0)
+        goto return_false;
+
+    WT_RET(__wt_config_gets_def(session, cfg, "next_random", 0, &cval));
+    if (cval.val != 0)
+        goto return_false;
+
+    WT_RET(__wt_config_gets_def(session, cfg, "readonly", 0, &cval));
+    if (cval.val)
+        goto return_false;
+
+    /* Checkpoints are readonly, we won't cache them. */
+    WT_RET(__wt_config_gets_def(session, cfg, "checkpoint", 0, &cval));
+    if (cval.val) {
+return_false:
+        *cacheablep = false;
+    } else {
+        *cacheablep = true;
+    }
+
+    return (0);
+}
+
+/*
+ * __cursor_reuse_or_init --
+ *     Initialization shared between reuse of a cached cursor and initialization of a new cursor.
+ *
+ * Flags set or cleared by this function: WT_CURSTD_APPEND, WT_CURSTD_DEBUG_RESET_EVICT.
+ *
+ * Flags set by this function: WT_CURSTD_RAW.
+ *
+ * Flags cleared by this functions: WT_CURSTD_CACHEABLE.
+ */
+static int
+__cursor_reuse_or_init(WT_SESSION_IMPL *session, WT_CURSOR *cursor, const char *cfg[],
+  bool *readonlyp, WT_CURSOR **ownerp, WT_CURSOR **cdumpp)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CURSOR *cdump;
+    WT_CURSOR *owner;
+
+    if (cfg != NULL) {
+        /*
+         * WT_CURSTD_APPEND is only relevant to column stores.
+         */
+        if (WT_CURSOR_RECNO(cursor)) {
+            WT_RET(__wt_config_gets_def(session, cfg, "append", 0, &cval));
+            if (cval.val != 0)
+                F_SET(cursor, WT_CURSTD_APPEND);
+            else
+                F_CLR(cursor, WT_CURSTD_APPEND);
+        }
+
+        /* WT_CURSTD_RAW */
+        WT_RET(__wt_config_gets_def(session, cfg, "raw", 0, &cval));
+        if (cval.val != 0)
+            F_SET(cursor, WT_CURSTD_RAW);
+
+        /* WT_CURSTD_DEBUG_RESET_EVICT */
+        WT_RET(__cursor_config_debug(cursor, cfg));
+    }
+
+    /* Readonly? */
+    WT_RET(__any_cursor_must_be_readonly(session, cfg, readonlyp));
+
+    /* WT_CURSTD_CACHEABLE */
+    if (*readonlyp) /* We do not cache read-only cursors. */
+        F_CLR(cursor, WT_CURSTD_CACHEABLE);
+
+    /*
+     * dump If an index cursor is opened with dump, then this function is called on the index files,
```

Comment:
np: capitalize dump

Reply:
I didn't add the debug comments. I didn't want to lose information. Some of them got moved due to restructuring. 

This comment is straight from the original code including the uncapitalized dump.
I see why dump is uncapitalized. It refers to opening a cursor with the dump configuration option. The dump configuration option is uncapitalized. Plus "dump" at the start matches "dump" in the text.

I changed the comment to: "If an index cursor is opened with the dump configuration option, ..."

Reply:
I just realized what you meant. By debug comments you meant the comments which flags are set, cleared, etc. by a function, not the comments like the dump comment above. I don't consider them debug comments. I considered they are summaries of what the function does. I planned to leave them. I can remove them if that is what you meant.

Reply:
Since Jie is on vacation I have removed the comments that I think he meant.

Jie: It would have helped if you had given line numbers for the comments that you considered debug comments.